From 4723e1754cd28657183358334fe77fe328f01bd0 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos.zoulas@twosigma.com>
Date: Fri, 18 Oct 2019 13:35:49 -0400
Subject: [PATCH] When the load of history caused an error, say so.
diff --git a/sh.err.c b/sh.err.c
index c676f5e..010b23c 100644
--- a/sh.err.c
+++ b/sh.err.c
@@ -43,6 +43,7 @@
 #endif
 
 char   *seterr = NULL;	/* Holds last error if there was one */
+extern int enterhist;
 
 #define ERR_FLAGS	0xf0000000
 #define ERR_NAME	0x10000000
@@ -630,6 +631,8 @@ stderror(unsigned int id, ...)
 	 */
 	flush();/*FIXRESET*/
 	haderr = 1;		/* Now to diagnostic output */
+	if (enterhist)
+	    xprintf("Can't load history: ");/*FIXRESET*/
 	if (flags & ERR_NAME)
 	    xprintf("%s: ", bname);/*FIXRESET*/
 	if ((flags & ERR_OLD)) {
